File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        while s:
            if "()" in s:
                s = s.replace("()","")
            elif "[]" in s:
                s = s.replace("[]", "")
            elif "{}" in s:
                s = s.replace("{}", "")
            else:
                return False
        

        return True
        
        # TC -> O(N)
        # SC -> O(N)

File: 20-valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        while(!s.isEmpty()){
            if(s.contains("()")){
                s = s.replace("()","");
            }else if(s.contains("{}")){
                s = s.replace("{}","");
            }else if(s.contains("[]")){
                s = s.replace("[]","");
            }else{
                return false;
            }
        }

        if(s.isEmpty()) return true;

        return false;
        
    }
}